### PR TITLE
Added check that verifies that there are no forbidden packages in use.

### PR DESCRIPTION
**Pull request by VelinYordanov**
See #297 
I made the check a bit more abstract and it works for all kinds of packages, not just guava.

Signed-off-by: VelinYordanov <velin.iordanov@gmail.com>

### CODE REVIEW

**Comment by VelinYordanov** [comment]
@martinvw Any remarks?

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/ForbiddenPackageUsageCheck.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * A check that verifies that there are no forbidden packages in use.
+ * 
+ * @author Velin Yordanov - initial contribution
+ *
+ */
+public class ForbiddenPackageUsageCheck extends AbstractCheck {
+    private final static String MESSAGE = "The package %s should not be used.";
+    private Collection<String> forbiddenPackages;
+    private Collection<String> exceptions;
+    private Map<String, Integer> importsToLineNumbers = new Hashtable<>();
```
Comment (martinvw) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/ForbiddenPackageUsageCheck.java]: A `Hashtable` is old stuff, why use it here?

File: sat-plugin/src/main/resources/rulesets/checkstyle/rules.xml [hunk 2]
```diff
@@ -196,7 +196,12 @@
       <property name="severity" value="warning" />
     </module>
     <module name="org.openhab.tools.analysis.checkstyle.InheritDocCheck">
+      <property name="severity" value="warning" />      
+    </module>
+    <module name="org.openhab.tools.analysis.checkstyle.ForbiddenPackageUsageCheck">
       <property name="severity" value="warning" />
+      <property name="forbiddenPackages" value="${checkstyle.forbiddenPackageUsageCheck.forbiddenPackages}" default="" />
```
Comment (martinvw) [inline sat-plugin/src/main/resources/rulesets/checkstyle/rules.xml]: Should it have a default? Same for the one above?

**Comment by dstivanov** [comment]
Thanks for your contributions!

**Pull request merged**